crates/polars-io/src/parquet/read/read_impl.rs

```diff
@@ -312,12 +315,18 @@ fn rg_to_dfs_prefiltered(
                 if let Some(rc) = &row_index {
                     df.with_row_index_mut(&rc.name, Some(rg.row_offset + rc.offset));
                 }
-                df = df.filter(mask)?;
+                // We don't want to parallelize here since we are already parallelizing over the
+                // row groups here.
+                df = df._filter_seq(mask)?;
```

Comment:
Isn't it better to have more tasks? :thinking:

Reply:
Won't recursive rayon calls lead to unnecessary contention and overhead?

Reply:
I will revert this. I know that we are turning off parallelization in `rg_to_dfs_par_over_rg` at the same point. I assumed for this reason.